framework/resource/internal/wrapper.go

```diff
@@ -1,7 +1,36 @@
 package internal
 
-import "github.com/giantswarm/operatorkit/framework"
+import (
+	old "github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/internal/framework"
+)
+
+type OldWrapper interface {
+	Underlying() old.Resource
+}
+
+func OldUnderlying(r old.Resource) old.Resource {
+	for {
+		wrapper, ok := r.(OldWrapper)
+		if ok {
+			r = wrapper.Underlying()
+		} else {
+			return r
+		}
+	}
+}
 
 type Wrapper interface {
-	Underlying() framework.Resource
+	Wrapped() framework.Resource
+}
+
+func Underlying(r framework.Resource) framework.Resource {
+	for {
+		wrapper, ok := r.(Wrapper)
+		if ok {
+			r = wrapper.Wrapped()
+		} else {
+			return r
+		}
+	}
```

Comment:
@xh3b4sd here's the reusable Underlying func. I didn't want to use Underlying name everywhere thinking that would be more confusing.